test/ibmq/test_ibmq_job.py

```diff
@@ -386,6 +386,38 @@ def test_retrieve_failed_job_device(self, provider):
         new_job = provider.backends.retrieve_job(job.job_id())
         self.assertTrue(new_job.error_message())
 
+    @requires_provider
+    def test_retrieve_failed_job_simulator_partial(self, provider):
+        """Test retrieving partial results from a simulator backend."""
+        backend = provider.get_backend('ibmq_qasm_simulator')
+
+        qc_new = transpile(self._qc, backend)
+        qobj = assemble([qc_new, qc_new], backend=backend)
+        qobj.experiments[1].instructions[1].name = 'bad_instruction'
+
+        job = backend.run(qobj)
+        result = job.result(partial=True)
+
+        circuit_results = result['results']
+        self.assertTrue(circuit_results[0]['success'])
+        self.assertFalse(circuit_results[1]['success'])
+
+    @requires_provider
+    def test_retrieve_failed_job_device_partial(self, provider):
```

Comment:
- if this is meant to run on a real backend, it needs to say `@slow_test` or `run_on_staging`
- docstring needs to be updated 
- real backend doesn't return partial results, so `result['results']` will give an empty list.